rpd_generator/bdl_structure/bdl_commands/zone.py

```diff
@@ -321,6 +327,58 @@ def populate_data_elements(self):
                     zone_ef_power_per_flow * exhaust_airflow
                 )
             return
+        # Populate Terminal for terminals_has_demand_control_ventilation
+        if self.keyword_value_pairs.get(BDL_ZoneKeywords.OA_FLOW_PER) is not None:
+            if (
+                self.parent.keyword_value_pairs.get(BDL_SystemKeywords.DOA_SYSTEM)
+                is not None
+            ):
+
+                if (
+                    self.parent.keyword_value_pairs.get(
+                        BDL_SystemKeywords.MIN_OA_METHOD
+                    )
+                    == BDL_SystemMinimumOutdoorAirControlOptions.DCV_ZONE_SENSORS
+                ):
+                    self.terminals_has_demand_control_ventilation[2] = True
+                elif (
+                    self.parent.keyword_value_pairs.get(
+                        BDL_SystemKeywords.DOAS_ATTACHED_TO
+                    )
+                    == BDL_DOASAttachedToOptions.AHU_MIXED_AIR
+                    and self.parent.keyword_value_pairs.get(
+                        BDL_SystemKeywords.MIN_OA_METHOD
+                    )
+                    == BDL_SystemMinimumOutdoorAirControlOptions.DCV_RETURN_SENSOR
+                ):
+                    self.terminals_has_demand_control_ventilation[2] = True
+                else:
+                    self.terminals_has_demand_control_ventilation[2] = False
+            else:
+                if (
+                    self.parent.keyword_value_pairs.get(
+                        BDL_SystemKeywords.MIN_OA_METHOD
+                    )
+                    == BDL_SystemMinimumOutdoorAirControlOptions.DCV_ZONE_SENSORS
+                    or self.parent.keyword_value_pairs.get(
+                        BDL_SystemKeywords.MIN_OA_METHOD
+                    )
+                    == BDL_SystemMinimumOutdoorAirControlOptions.DCV_RETURN_SENSOR
+                ):
+                    self.terminals_has_demand_control_ventilation[0] = True
+                else:
+                    self.terminals_has_demand_control_ventilation[0] = False
+        # No per person rate was populated, set equal to false if the terminal type exists
+        else:
+            if (
+                self.parent.keyword_value_pairs.get(BDL_SystemKeywords.DOA_SYSTEM)
+                is not None
+            ):
+                self.terminals_has_demand_control_ventilation[2] = False
```

Comment:
relocate to block of code starting on line 238 with comment `# Populate DOAS Terminal data elements if applicable`